apptail/storage/tail_tracker.go

```diff
@@ -0,0 +1,173 @@
+package storage
+
+import (
+	"fmt"
+	"github.com/ActiveState/log"
+	"runtime"
+	"sync"
+	"sync/atomic"
+	"time"
+)
+
+type Tracker interface {
+	Update(instKey string, childKey string, childVal int64)
+	LoadTailers()
+	Remove(key string)
+	RegisterInstance(instKey string)
+	InitializeChildNode(instKey string, childkey string, offSet int64)
+	Submit()
+	StartSubmissionTimer(retentionPeriod time.Duration)
+	IsInstanceRegistered(instKey string) bool
+	IsChildNodeInitialized(instKey string, childkey string) bool
+	GetFileCachedOffset(instkey string, fname string) int64
+}
+
+type BoxedInt64 struct{ V int64 }
```

Comment:
Err? I don't see that near [FileStorage.Write()](https://github.com/ActiveState/logyard-apps/pull/7/files#diff-0abd8b982345db71cd3b8ae245d05e6cR28).